src/app/meme-card/meme-card.component.ts

```diff
@@ -0,0 +1,62 @@
+import { Component, OnInit, Input } from '@angular/core';
+
+@Component({
+  selector: 'app-meme-card',
+  templateUrl: './meme-card.component.html',
+  styleUrls: ['./meme-card.component.scss']
+})
+
+export class MemeCardComponent implements OnInit {
+  @Input() cardHeight: string;
+  @Input() cardWidth: string;
+  @Input() username: string;
+
+  // give default value to height and width.
+
+  // if cardHeight is > 0, then change CSS
+
+  // if cardWidth is > 0, then change CSS
+
+  voteCount = 0;
+  voted = false;
+
+  // default style
+  cardStyle = {
```

Comment:
In the future if you want to declare like this and overwrite later without worries of type errors, you can do `cardStyle: {};` or `cardStyle: any` (but any is not the best idea...)